flake.nix

```diff
@@ -0,0 +1,209 @@
+{
+  description = "rosettaboy nix flake";
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-22.11;
+    flake-utils.url = github:numtide/flake-utils;
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    gitignore = {
+      url = "github:hercules-ci/gitignore.nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    gomod2nix-src = {
+      url = "github:nix-community/gomod2nix";
+      flake = false;
+    };
+    nim-argparse = {
+      url = "github:iffy/nim-argparse";
+      flake = false;
+    };
+    php-sdl = {
+      url = "github:Ponup/php-sdl";
+      flake = false;
+    };
+    naersk = {
+      url = "github:nix-community/naersk";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    zig-overlay = {
+      url = "github:mitchellh/zig-overlay/17352071583eda4be43fa2a312f6e061326374f7";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    zig-sdl = {
+      url = "github:MasterQ32/SDL.zig/6a9e37687a4b9ae3c14c9ea148ec51d14e01c7db";
+      flake = false;
+    };
+    zig-clap = {
+      url = "github:Hejsil/zig-clap/e5d09c4b2d121025ad7195b2de704451e6306807";
+      flake = false;
+    };
+    gb-autotest-roms = {
+      url = "github:shish/gb-autotest-roms";
+      flake = false;
+    };
+    cl-gameboy = {
+      url = "github:sjl/cl-gameboy";
+      flake = false;
+    };
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+    flake-compat,
+    gitignore,
+    gomod2nix-src,
+    nim-argparse,
+    php-sdl,
+    naersk,
+    zig-overlay,
+    zig-sdl,
+    zig-clap,
+    gb-autotest-roms,
+    cl-gameboy
+  }: flake-utils.lib.eachDefaultSystem (system: let
+    pkgs = nixpkgs.legacyPackages.${system};
+    lib = pkgs.lib;
+    inherit (builtins) mapAttrs;
+    inherit (lib) hiPrio filterAttrs;
+    inherit (gitignore.lib) gitignoreSource;
+    gomod2nix' = rec {
+      gomod2nix = pkgs.callPackage "${gomod2nix-src}" { inherit (lib) buildGoApplication mkGoEnv; };
+      lib = pkgs.callPackage "${gomod2nix-src}/builder" { inherit gomod2nix; };
+    };
```

Comment:
I was trying to avoid using an overlay if possible.

See: https://github.com/nix-community/poetry2nix/commit/05b788f7b9beaa9014c9d794acf004e23ad571af
See: https://github.com/nix-community/gomod2nix/issues/102

Reply:
I saw your `gomod2nix` issue; as zimbatm notes in their blog post, I think it makes a lot of sense to be wary of using overlays in flakes that other people are likely to depend on but less so for flakes that are just providing packages to users. I think `rosettaboy` falls pretty comfortably into the latter category; it seems unlikely that many users will pull `rosettaboy` into their own flakes.

I don't feel strongly about this though; whatever approach you prefer is fine with me.